packages/core/src/model/circle-pack.ts

```diff
@@ -28,7 +28,7 @@ export class CirclePackChartModel extends ChartModel {
 		updateLegendAdditionalItems(options, zoomOptions)
 
 		const depth = this.getHierarchyLevel()
-		const userProvidedDepth = getProperty(options, 'circlePack', 'hierarchyLevel')
```

Comment:
If you looked at functions like getHierarchyLevel(), you'll see that they expected a `depth` property - not hierarchyLevel.